cosmo/monitors/nuv_dark_monitor.py

```diff
@@ -59,19 +64,33 @@ def calculate_dark_rate(self, dataframe, xlim, ylim):
 
         counts = np.histogram(dataframe["TIME"], bins=time_bins)[0]
         npix = float((xlim[1] - xlim[0]) * (ylim[1] - ylim[0]))
-        dark_rate = np.median(counts / npix / timestep)
-        # what is going on with the whole histogram, ok to use median?
-
-        return dark_rate
+        dark_rate_array = counts / npix / timestep
+        # save the whole histogram in time bins, and then plot each of them
+
+        # make a decimal year array corresponding to the time bins of the
+        # dark rates
+        # do this with the expstart (mjd) and time array from the timeline
+        # extension
+        mjd_conversion_factor = 1.15741e-5
+        # taking the expstart, binning the time array by the timestep,
+        # removing the last element in the array (bin by front edge),
+        # and then multiplying by the conversion factor
+        mjd_array = dataframe["EXPSTART"] + dataframe["TIME_3"][::timestep][
+                                            :-1] * mjd_conversion_factor
+        mjd_array = Time(mjd_array, format='mjd')
+        dec_year_array = mjd_array.decimalyear
+
+        return dark_rate_array, dec_year_array
 
     def track(self):
         # track something. perhaps current dark rate?
```

Comment:
This is a great idea! That way you'd have that ~ number ~ needed for ETC updates right there and you wouldn't necessarily have to go through figures to figure it out.

Reply:
yeah once i get the histograms set that's what i'm going to put there